content/blog/query-kubernetes/index.md

```diff
@@ -0,0 +1,138 @@
+---
+title: Introducing CloudQuery
+authors: ["joe-duffy"]
+tags: ["Pulumi-News", "Kubernetes"]
+meta_desc: "Today we announced Pulumi CloudQuery, an SDK for programmatically querying cloud resoures."
+date: "2019-11-14"
+meta_image: "pulumi-crosswalk-k8s.png"
+---
+
+Today, as part of our Crosswalk for Kubernetes announcement[link], we're introducing CloudQuery for
+Kubernetes.
+
+CloudQuery is a relational TypeScript SDK for querying live Kubernetes resources. It makes it easy
+to answer questions that would normally be answered by hand-scripting calls to `kubectl`. Questions
+like:
+
+* How many distinct versions of MySQL are running in my cluster?
+* Which Pods are scheduled on nodes with high memory pressure?
+* Which Pods are publicly exposed to the internet via a load-balanced Service?
+
+CloudQuery supports both batch and streaming query modes. Batch queries run on a finite set of
+resources. But streaming queries operate on an unbounded stream of resource updates, making it easy
+to watch Kubernetes resources and to take action when specific events occur. For example:
+
+* Post a slack notification when we run out of disk space.
+* Page someone when a rollout fails.
+
+In this post, we'll look at both batch and streaming query modes, showing examples of each. The SDK
+repository also contains a set of example queries, including instructions for how to run them.
+
+## Batch queries
+
+Below is one of our simplest examples, a query program that finds all distinct versions of MySQL
+running in the default namespace of the active cluster of your `$KUBECONFIG` file.
+
+The core primitive of batch queries is `list`. In this case, `list("v1", "Pod")` causes us to list
+all Pods in the default namespace. The rest of the query "flattens" this list of Pods into a list of
+all container images, filters out all images that don't contain the string "mysql", and filters that
+down to only distinct image names.
+
+```typescript
+import * as kq from "@pulumi/query-kubernetes";
+
+// Find all distinct versions of MySQL running in your cluster.
+const mySqlVersions = kq
+    .list("v1", "Pod")
+    .flatMap(pod => pod.spec.containers)
+    .map(container => container.image)
+    .filter(imageName => imageName.includes("mysql"))
+    .distinct();
+
+mySqlVersions.forEach(console.log);
+```
+
+Query programs are run via the `pulumi query` command. When run, output looks like the following:
```

Comment:
Do we need to note here that this command is still `PULUMI_EXPERIMENTAL`?  More generally, do we need to provide a link to a guide for installing and using this?